### PR TITLE
Disconnect expired k8s clients, fixes #2618

### LINKED ISSUE #2618: Teleport Kubernetes should respect idle_timeout and disconnect_expired_cert
## Description

Teleport Kubernetes proxy should disconnect expired certs when specified and respect idle timeout to be consistent with Teleport SSH.

### PR DESCRIPTION
**Pull request by klizhentas**
This commit implements disconnects of kubernetes
clients with expired certificates and terminates
idle sessions, according to 'disconnect_expired_cert'
and 'client_idle_timeout' existing settings making
the behavior consistent with SSH sessions.

### CODE REVIEW

File: lib/kube/proxy/forwarder.go [hunk 1]
```diff
@@ -213,19 +220,26 @@ func (f *Forwarder) Close() error {
 // authContext is a context of authenticated user,
 // contains information about user, target cluster and authenticated groups
 type authContext struct {
+	auth.AuthContext
+	kubeGroups    []string
+	cluster       cluster
+	clusterConfig services.ClusterConfig
+	// clientIdleTimeout sets information on client idle timeout
+	clientIdleTimeout time.Duration
+	// disconnectExpiredCert time.Time
```
Comment (r0mant) [inline lib/kube/proxy/forwarder.go]: Invalid godoc.

File: lib/srv/monitor.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	"time"
+
+	"github.com/gravitational/teleport/lib/events"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	log "github.com/sirupsen/logrus"
+)
+
+// ActivityTracker is activity tracker
```
Comment (r0mant) [inline lib/srv/monitor.go]: Nit: This godoc could use a little more love :)

File: lib/srv/monitor.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+type ActivityTracker interface {
+	// GetClientLastActive returns the time of the last recorded activity
+	GetClientLastActive() time.Time
+	// UpdateClient
```
Comment (r0mant) [inline lib/srv/monitor.go]: Missing/non-matching godoc.

File: lib/srv/monitor.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+				events.Reason:          fmt.Sprintf("client certificate expired at %v", w.Clock.Now().UTC()),
+			}
+			w.Audit.EmitAuditEvent(events.ClientDisconnectEvent, event)
+			w.Entry.Debugf("Disconnecting client: %v", event[events.Reason])
+			w.Conn.Close()
+			return
+		case <-idleTime:
+			now := w.Clock.Now().UTC()
+			clientLastActive := w.Tracker.GetClientLastActive()
+			w.Entry.Debugf("client last active %v, client idle timeout %v", clientLastActive, w.ClientIdleTimeout)
```
Comment (r0mant) [inline lib/srv/monitor.go]: Nit: client -> Client

**Review by r0mant — APPROVED** [review]

File: lib/srv/ctx.go [hunk 5]
```diff
@@ -264,12 +264,29 @@ func NewServerContext(srv Server, conn *ssh.ServerConn, identityContext Identity
 	})
 
 	if !ctx.disconnectExpiredCert.IsZero() || ctx.clientIdleTimeout != 0 {
-		go ctx.periodicCheckDisconnect()
```
Comment (russjones) [inline lib/srv/ctx.go]: I don't think anyone else is calling this function, so you can remove the function as well.

**Review by russjones — APPROVED** [review]

**Pull request merged**